Makefile

```
@@ -68,7 +78,7 @@ HAS_SETUP_ENVTEST := $(shell command -v setup-envtest;)
 testall: manifests generate fmt vet golangci-lint test ## Run tests.
 
 test: envtest
-	KUBEBUILDER_ASSETS="$(shell setup-envtest --arch=amd64 use $(ENVTEST_K8S_VERSION) -p path)" go test ./... -coverprofile cover.out
+	KUBEBUILDER_ASSETS="$(shell setup-envtest --arch=${ARCH} --os=${OS} use $(ENVTEST_K8S_VERSION) -p path)" go test ./... -coverprofile cover.out
```

```suggestion
	KUBEBUILDER_ASSETS="$(shell setup-envtest use $(ENVTEST_K8S_VERSION) -p path)" go test ./... -coverprofile cover.out
```

Until K8s v1.23, setup-envtest didn't support the arm platform. So we explicitly specify the amd64 platform.

We should remove a `--arch` flag since the tools supported the arm64 platform since k8s v1.24.

https://github.com/kubernetes-sigs/kubebuilder/issues/2664#issuecomment-1149956592

---

Thanks, I known it now. So remove the `--arch` and `--os` at all? But the default value is arm64 and darwin.
`--arch string                 architecture to download for (e.g. amd64, for listing operations, use '*' to list all platforms) (default "arm64")`
`--os string                   os to download for (e.g. linux, darwin, for listing operations, use '*' to list all platforms) (default "darwin")`

---

Yes, we should remove them since setup-envtest automatically detects the environment using `GOOS` and `GOARCH`.